lightning/src/ln/channel.rs

```diff
@@ -2962,6 +2971,41 @@ impl<Signer: Sign> Channel<Signer> {
 			panic!("Cannot update fee while peer is disconnected/we're awaiting a monitor update (ChannelManager should have caught this)");
 		}
 
+		// Before proposing a feerate update, check that we can actually afford the new fee.
+		let inbound_stats = self.get_inbound_pending_htlc_stats(Some(feerate_per_kw));
+		let outbound_stats = self.get_outbound_pending_htlc_stats(Some(feerate_per_kw));
+		// In case of a concurrent update_add_htlc proposed by our counterparty, we might
+		// not have enough balance value remaining to cover the onchain cost of this new
+		// HTLC weight. If this happens, our counterparty fails the reception of our
+		// commitment_signed including this new HTLC due to infringement on the channel
+		// reserve.
+		// To prevent this case, we compute our outbound update_fee with an HTLC buffer of
+		// size 2. However, if the number of concurrent update_add_htlc is higher, this still
```

Comment:
Maybe somewhat unorthodox idea - what if we just use the maximum HTLC count from our counterparty here? Make sure we can always afford for our counterparty to add max-htlcs plus whatever HTLCs we currently have pending. I guess that'd be a larger change across the state machine not just in update_fee.

Reply:
At first thought, I would say using the maximum HTLC count from our counterparty would be super restrictive in the sense of generating a far-higher rate of fail-to-increase-feerate at the advantage, I concede, to decrease our rate of unexpected closes because of non-affordable concurrent inbound HTLC ? And I think making the assumption that you have 1 or 2 concurrent inbound HTLC is more realistic with today network than `MAX_ACCEPTED_HTLCS` in-flight ?

Reply:
Yea, maybe? I was just looking at my node last night and watched someone do, I assume, payment probes 10-or-so at a time. Anyway, this conversation maybe should be separate, as its not specific to just this location/the noew code.